taichi/backends/opengl/codegen_opengl.cpp

```diff
@@ -106,6 +106,7 @@ class KernelGen : public IRVisitor {
   bool is_top_level_{true};
   CompiledProgram compiled_program_;
   UsedFeature used;  // TODO: is this actually per-offload?
+  int num_arr = static_cast<int>(GLBufId::Arr);
```

Comment:
Hmm, why this cast?

Reply:
No serious reason really, mainly because I want to use `++` operator on it below and it's easier to do with int :P 
btw `num_arr` wasn't an accurate name for this either so I updated it as well. :D